### PR TITLE
enhance:Add integration test for searching when coord down

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #29772

### CODE REVIEW

File: tests/integration/coorddownsearch/search_after_coord_down_test.go [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+	"github.com/milvus-io/milvus/tests/integration"
+)
+
+type CoordDownSearch struct {
+	integration.MiniClusterSuite
+}
+
+const (
+	Dim                         = 128
+	numCollections              = 500
```
Comment (yiwangdr) [inline tests/integration/coorddownsearch/search_after_coord_down_test.go:50]: 500 is unnecessary. 
numCollections = 1 is good enough.

**Review by yiwangdr — APPROVED** [review]

File: tests/integration/coorddownsearch/search_after_coord_down_test.go
Comment (yiwangdr) [inline tests/integration/coorddownsearch/search_after_coord_down_test.go:1]: Let's do a followup PR to move the duplicate data setup functions into a util file.

File: tests/integration/coorddownsearch/search_after_coord_down_test.go [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+	s.NoError(err)
+
+	err = merr.Error(createCollectionStatus)
+	s.NoError(err)
+
+	showCollectionsResp, err := c.Proxy.ShowCollections(context.TODO(), &milvuspb.ShowCollectionsRequest{})
+	s.NoError(err)
+	s.True(merr.Ok(showCollectionsResp.GetStatus()))
+
+	batchSize := 500000
```
Comment (xiaofan-luan) [inline tests/integration/coorddownsearch/search_after_coord_down_test.go:81]: might be too much? 5000 data is enough/

File: tests/integration/coorddownsearch/search_after_coord_down_test.go [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+	s.NotEmpty(segments)
+	s.WaitForFlush(context.TODO(), ids, flushTs, dbName, collectionName)
+	log.Info("=========================Data flush finished=========================")
+
+	// create index
+	createIndexStatus, err := c.Proxy.CreateIndex(context.TODO(), &milvuspb.CreateIndexRequest{
+		CollectionName: collectionName,
+		FieldName:      integration.FloatVecField,
+		IndexName:      "_default",
+		ExtraParams:    integration.ConstructIndexParam(dim, integration.IndexFaissIvfFlat, metric.IP),
```
Comment (xiaofan-luan) [inline tests/integration/coorddownsearch/search_after_coord_down_test.go:125]: create a flat index

**Comment by xiaocai2333** [comment]
@xiaofan-luan @czs007 
Pay special attention to:
1. for requests with `Bounded`  consistency level, can't set a local timestamp for guarantee timestamp, it will wait for tsafe until the timestamp. When RootCoord goes down, it will keep waiting for tsafe because the tsafe won't be updated.
2. When RootCoord goes down, try to search with `strong` consistency level will cost about 10 senconds due to retry logic.

**Comment by xiaocai2333** [comment]
rerun ut

**Pull request merged**